### PR TITLE
Add basic unit tests and minor refactoring

### PR DESCRIPTION
**Pull request by alexander-demichev**
This PR introduces basic unit tests for testing out config migration logic.

### CODE REVIEW

File: ocp4crd/oauth/oauth_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+	ocp3 "github.com/fusor/cpma/ocp3config"
+)
+
+func TestGenerateMasterConfig(t *testing.T) {
+	testConfig := ocp3.Config{
+		Masterf: "../../examples/source/etc/origin/master/master-config.yaml",
+	}
+	masterConfig := testConfig.ParseMaster()
+
```
Comment (alexander-demichev) [inline ocp4crd/oauth/oauth_test.go]: Defining identity providers manually took some time, but I would like to be 146% sure our generated YML is valid.

File: ocp4crd/oauth/oauth.go [hunk 2]
```diff
@@ -88,48 +88,64 @@ func Generate(masterconfig configv1.MasterConfig) error {
...
 			crd.Spec.IdentityProviders = append(crd.Spec.IdentityProviders, idP)
 		default:
 			log.Println("can't handle: ", kind)
 		}
 
 	}
 
+	return &crd, nil
+}
+
```
Comment (alexander-demichev) [inline ocp4crd/oauth/oauth.go]: Moved some logic to separate functions. Should be more readable now.

File: ocp3config/ocp3_test.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+				},
+				configv1.IdentityProvider{
+					Name: "github123456789",
+				},
+			},
+		},
+	}
+
+	resMasterConfig := testConfig.ParseMaster()
+
```
Comment (alexander-demichev) [inline ocp3config/ocp3_test.go]: I decided to put some random fields values in out test object, assuming there is no need to check all of them.

File: ocp3config/ocp3.go [hunk 4]
```diff
@@ -24,10 +24,10 @@ import (
 // Config represents OCP3 configuration
 type Config struct {
 	master  configv1.MasterConfig
-	masterf string
+	Masterf string
```
Comment (alexander-demichev) [inline ocp3config/ocp3.go]: Made this fields public, so I could set them in tests.

File: cmd/root.go [hunk 5]
```diff
@@ -46,7 +46,13 @@ var rootCmd = &cobra.Command{
 
 		m := ocp3config.ParseMaster()
 
-		oauth.Generate(m)
+		crd, err := oauth.Generate(m)
```
Comment (alexander-demichev) [inline cmd/root.go]: If Generate returns CRD, it's more easy to test

**Comment by alexander-demichev** [comment]
[[ControlPlane] Create a model for how to unit test CPMA utility. Begin with OAuth provider translations.](https://trello.com/c/ij4yMW3U/86-controlplane-create-a-model-for-how-to-unit-test-cpma-utility-begin-with-oauth-provider-translations)

**Comment by alexander-demichev** [comment]
[[ControlPlane] Create a model for how to unit test CPMA utility. Begin with OAuth provider translations.](https://trello.com/c/ij4yMW3U/86-controlplane-create-a-model-for-how-to-unit-test-cpma-utility-begin-with-oauth-provider-translations)

File: ocp3config/ocp3_test.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import (
+	"testing"
+
+	configv1 "github.com/openshift/api/legacyconfig/v1"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConfigParseMaster(t *testing.T) {
+	testConfig := Config{
+		Masterf: "../examples/source/etc/origin/master/master-config.yaml",
```
Comment (arapov) [inline ocp3config/ocp3_test.go]: I guess we want to have test data input along the code, somewhere in ocp3config directory.

File: ocp4crd/oauth/oauth.go [hunk 7]
```diff
@@ -88,48 +88,64 @@ func Generate(masterconfig configv1.MasterConfig) error {
 	for _, p := range auth.IdentityProviders {
 		p.Provider.Object, _, err = serializer.Decode(p.Provider.Raw, nil, nil)
```
Comment (arapov) [inline ocp4crd/oauth/oauth.go]: remove empty line before check for nil

File: ocp4crd/oauth/oauth.go [hunk 8]
```diff
@@ -88,48 +88,64 @@ func Generate(masterconfig configv1.MasterConfig) error {
...
-			var htpasswd configv1.HTPasswdPasswordIdentityProvider
-			_, _, _ = serializer.Decode(p.Provider.Raw, nil, &htpasswd)
-
-			idP.Type = "HTPasswd"
-			idP.Challenge = p.UseAsChallenger
-			idP.Login = p.UseAsLogin
-			idP.MappingMethod = p.MappingMethod
-			idP.HTPasswd.FileData.Name = htpasswd.File
-
+			idP := buildHTPasswdIP(serializer, p)
```
Comment (arapov) [inline ocp4crd/oauth/oauth.go]: This change has nothing to do with the commit implementing unit test, therefore it must be in separate commit.

File: README.md [hunk 9]
```diff
@@ -1,2 +1,35 @@
...
+--debug                   show debug ouput
+-h, --help                help for cpma
+-o, --output-dir string   set the directory to store extracted configuration.
+```
+
+You can find example config in `examples/` 
+
+Example:
+
+```console
```
Comment (arapov) [inline README.md]: command line or cli.

Comment (alexander-demichev) [reply README.md]: This is an annotation, it won't be rendered

File: ocp3config/ocp3_test.go (hunk 6 above)
Comment (alexander-demichev) [reply ocp3config/ocp3_test.go]: How do see it?
